added new functional

JaysonJuza opened the pull request:
created a Spring Boot project with pom.xml file with all required dependencies
added Book entity
added BookRepository
added BookService
customized database access
customizedcheckstyle
customized application.properties in the test/resources folder
customized CI (Continuous Integration)

kerrrusha commented on .mvn/wrapper/maven-wrapper.jar:
dont push maven-related files

kerrrusha commented on src/main/java/onlinebookstoreapp/model/Book.java:
```diff
@@ -0,0 +1,27 @@
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity(name = "books")
```
```suggestion
@Entity("books")
```

kerrrusha commented on src/main/java/onlinebookstoreapp/model/Book.java:
```diff
@@ -0,0 +1,27 @@
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity(name = "books")
+@Data
```
Using Lombok's `@Data` annotation on `@Entity` classes in Java, particularly in JPA (Java Persistence API) environments like Hibernate, is generally not recommended due to potential issues with equals, hashCode, and toString methods.
When you use `@Data`, Lombok generates these methods automatically, which can lead to problems in JPA environments. For instance, including all fields in equals and hashCode methods might lead to unexpected behavior when entities are managed by JPA. Additionally, the toString method might inadvertently trigger lazy loading of entity associations, causing performance issues or even unexpected behavior.
In JPA, it's often recommended to manually implement equals, hashCode, and toString methods in entity classes, customizing them based on business requirements and ensuring they don't interfere with the persistence mechanism. Therefore, while Lombok can save boilerplate code in many scenarios, it's best to avoid using `@Data` on JPA entity classes and instead selectively use Lombok annotations like `@Getter`, `@Setter`, `@EqualsAndHashCode`, and `@ToString` with caution and understanding of their implications in JPA environments.

MAMentorFX replied:
Forget abuut @Data for entities, @Data consists of @EqualsAndHashCode and @ToString, those two may lead to Lazy loading exception. Use @Getter and @Setter instead

kerrrusha commented on src/main/java/onlinebookstoreapp/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,55 @@
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
use `@RequiredArgsConstructor` instead
```suggestion
```

kerrrusha commented on src/main/java/onlinebookstoreapp/service/impl/BookServiceImpl.java:
```diff
@@ -0,0 +1,28 @@
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
same

JaysonJuza committed:
added new functional

MAMentorFX commented on mvnw:
```diff
@@ -0,0 +1,308 @@
+#!/bin/sh
```
mvnw and mvnw.cmd should  be in gitignore file

MAMentorFX replied:
Not fixed.
Also .mvn/wrapper in gitignore.

MAMentorFX commented on src/main/java/onlinebookstoreapp/model/Book.java:
```diff
@@ -0,0 +1,27 @@
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false)
+    private BigDecimal isbn;
```
shoudl be unique

MAMentorFX replied:
fixed

JaysonJuza committed:
added mvnw and mvnw.cmd into .gitignore file.
Change annotation @Data to @Setter, @Getter.
Added properties for variable param: unique

MAMentorFX reviewed (changes requested):
Put mvn related files to gitignore

JaysonJuza committed:
added .mvn/wrapper into .gitignore file.
Added properties for variable author: unique = true

MentorMA reviewed (changes requested):
resolve problem with git ignore files, clear redundant files from repository

JaysonJuza committed:
clear redundant files from repository

JaysonJuza committed:
added and customized Book Controller,
added BookDto and CreateBookRequestDto objects for the Book entity and for use at the service level,
added the mapstruct library and added the corresponding dependency to pom.xml,
added MapperConfig configuration class ,
added BookMapper for converting entity to DTO object and vice versa,
custom exception added

JaysonJuza committed:
change rollback
